<launch>
  <param name="robot_type" value="ur3" />

  <arg name="simulated" default='true'/>
  <arg name="rik_use_julia" default='false'/>
  <arg name="limited" default='false' />

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <param name="$robot_description" command="$(find xacro)/xacro --inorder '$(find robot_descriptions)/urdf/universal_robots/relaxed_ik_ur3_robotiq_85.urdf.xacro'" />

  <!-- If simulated, we do not have a robot connected -->
  <group if="$(eval arg('simulated') == true)">
    <param name="robot_connected" value="false" />

    <node name="virtual_robot_controller" pkg="robot_behavior" type="virtual_robot_controller.py">
      <param name="~gripper_joint" value="robotiq_85_left_knuckle_joint" />
    </node>

    <!-- We do not have a robot connected, so publish fake joint states -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="/use_gui" value="false"/>
      <rosparam param="/source_list">[/virtual_controller/joint_states]</rosparam>
    </node>

    <!-- Given the published joint states, publish tf for the robot links -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
  </group>


  <!-- Else, we launch physical robot -->
  <group if="$(eval arg('simulated') == false)">
    <param name="robot_connected" value="true" />

    <!-- Launch robot driver -->
    <!-- TODO -->

    <!-- Launch gripper driver -->
    <include file="$(find m1n6s_robotiq85_moveit_config)/launch/robotiq_85.launch">
      <arg name="dual_gripper" value="false"/>
      <arg name="comport" value="/dev/ttyUSB0"/>
      <arg name="baud" value="115200"/>
      <arg name="run_test" value="false"/>
    </include>

    <!-- Run gripper command action server, which receives planned motion goal under "gripper_command". After receiving the command from Moveit Rviz Plugin, this node act as an actionlib client and send the goal to the finger position action server from kinova_driver.  -->
    <node name="gripper_action_server" pkg="m1n6s_robotiq85_moveit_config" type="gripper_action_server.py" output="screen" respawn="false"/>

  </group>


  <!-- Setup robot configuration for Relaxed Ik -->
  <param name="path_to_relaxed_ik_src" value="$(find relaxed_ik)/src"/>
  <param name="info_file_name" value="ur3_info.yaml"/>
  <param name="ee_link" value="tool0"/>

  <!-- RVIZ -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find iter_app)/rviz/rik-robot-tf.rviz"/>

</launch>
